sbol2/componentdefinition.py

```diff
@@ -227,62 +354,124 @@ def updateSequence(self, composite_sequence=""):
         raise NotImplementedError("Not yet implemented")
 
     def getInSequentialOrder(self):
-        """Orders this ComponentDefinition's member Components
-        into a linear arrangement based on Sequence Constraints.
+        """Get the primary sequence of a design in terms of its sequentially ordered
+        Components.
 
-        :return: Primary sequence structure.
+        :return: A list of Components.
         """
-        raise NotImplementedError("Not yet implemented")
+        return self.getPrimaryStructureComponents()
 
-    def hasUpstreamComponent(self, current_component):
+    def hasUpstreamComponent(self, component):
```

Comment:
Why does `testHasUpstreamComponent` still fail despite this method being implemented? (see test_component_definition.py around line 134.

Reply:
Still investigating this, but it looks like this test uses the CRISPR_EXAMPLE file.  I suspect the problem is with the file, rather than the `hasUpstreamComponent` method.  Note that TestAssemblyRoutines.test_inspect_primary_structure also tests this method

Reply:
This has been moved to #246